src/tasks/migration/utils.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as fse from "fs-extra";
+import * as path from "path";
+import { debounce } from "lodash";
+import { tasks, window, workspace } from "vscode";
+import { DeprecatedExportJarTaskProvider, ExportJarTaskProvider } from "../../exportJarSteps/ExportJarTaskProvider";
+import { contextManager } from "../../contextManager";
+import { Context } from "../../constants";
+import { DiagnosticProvider } from "./DiagnosticProvider";
+import { CodeActionProvider } from "./CodeActionProvider";
+
+export async function getTasksJsonPaths(): Promise<string[]> {
+    const tasksJsonPaths = [];
+    if (workspace.workspaceFolders?.length) {
+        for (const folder of workspace.workspaceFolders) {
+            const tasksJsonPath = path.join(folder.uri.fsPath, ".vscode", "tasks.json");
+            if (await fse.pathExists(tasksJsonPath)) {
+                tasksJsonPaths.push(tasksJsonPath);
+            }
+        }
+    }
+    return tasksJsonPaths;
+}
+
+export function asyncDebounce(func: any, wait: any, bind: any) {
```

Comment:
Why we need to wrapper it to async?

Reply:
it seems async function is not proper debounced from lodash debounce library. here I use the workaround here: https://github.com/lodash/lodash/issues/4815#issuecomment-859379489

Reply:
But we are not awaiting the function returned value. I think we can just kick off the execution without awaiting it. Do I miss anything here?

Reply:
get it. we can directly use debounce here.